lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -448,6 +448,35 @@ defmodule KernelTest do
     end
   end
 
+  test "delete_in/2" do
+    users = %{"john" => %{age: 27}, "meg" => %{age: 23}}
+
+    assert delete_in(users, ["john", :age]) ==
+           %{"john" => %{}, "meg" => %{age: 23}}
+
+    assert_raise FunctionClauseError, fn ->
+      delete_in(users, [])
+    end
+
+    assert_raise ArgumentError, "could not put/update key \"john\" on a nil value", fn ->
```

Comment:
This error message is not ideal as it appears the function being called is an update one, not a delete. However, because I am repurposing `get_and_update_in` for part of the `delete_in` function we get back its error messages. I'm happy to try and address if anyone has a suggestion.